src/shared/components/FileDrop/FileDrop.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useCallback, useState } from 'react'
+import { DropzoneOptions, useDropzone } from 'react-dropzone'
+import { CSSTransition, SwitchTransition } from 'react-transition-group'
+import Icon from '../Icon'
+import {
+  ButtonsGroup,
+  DismissButton,
+  DragAndDropArea,
+  DragDropText,
+  ErrorContainer,
+  ErrorIcon,
+  ErrorText,
+  InnerContainer,
+  Paragraph,
+  ProgressBar,
+  StyledIcon,
+  Thumbnail,
+  Title,
+  UploadButton,
+} from './FileDrop.style'
+import { Step } from '../MultiFileSelect'
+
+export type FileDropProps = {
+  step: Step
+  onUploadFile: (file: File) => void
+  icon: 'video-dnd' | 'image-dnd'
+  title: string
+  paragraph: string
+  thumbnail?: string
+  loading?: boolean
+  progress?: number
+  onReAdjustThumbnail: () => void
+}
+
+const FileDrop: React.FC<FileDropProps> = ({
+  onUploadFile,
+  step,
+  icon,
+  title,
+  paragraph,
+  thumbnail,
+  onReAdjustThumbnail,
+  progress,
+}) => {
+  const [error, setError] = useState<string>()
+
+  const onDropAccepted: DropzoneOptions['onDropAccepted'] = useCallback(
+    (acceptedFiles) => {
+      const [file] = acceptedFiles
+      onUploadFile(file)
+    },
+    [onUploadFile]
+  )
+
+  const onDropRejected: DropzoneOptions['onDropRejected'] = useCallback(() => {
+    setError('Wrong file type!')
```

Comment:
Does it reject file based only on type? Or e.g. size as well? Maybe we should be able to customise error message